src/achievements.cpp

```diff
@@ -0,0 +1,116 @@
+/*
+	Copyright (C) 2003 - 2022
+	by David White <dave@whitevine.net>
+	Part of the Battle for Wesnoth Project https://www.wesnoth.org/
+
+	This program is free software; you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation; either version 2 of the License, or
+	(at your option) any later version.
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY.
+
+	See the COPYING file for more details.
+*/
+
+#include "achievements.hpp"
+
+#include "filesystem.hpp"
+#include "game_config.hpp"
+#include "log.hpp"
+#include "preferences/general.hpp"
+#include "serialization/parser.hpp"
+#include "serialization/preprocessor.hpp"
+
+static lg::log_domain log_config("config");
+#define ERR_CONFIG LOG_STREAM(err, log_config)
+
+// TODO: horizontal scrollbar is present despite `horizontal_scrollbar_mode = "never"`
```

Comment:
The horizontal scrollbar you're seeing (probably) isn't something defined in the dialog file. There's a scrollbar in the window's widget definition which ensures that a dialog that's too wide can still be fully seen. If the scrollbar is scrolling the listbox's position and not just its content, then it's the scrollbar from the window definition.

Reply:
Huh, alright. It seems to be present whenever the vertical scrollbar is present and can only scroll right or left by a tiny amount (less than the width of the vertical scrollbar.

Reply:
Yeah, to be clear, I'm not saying we don't need to fix it, rather that `horizontal_scrollbar_mode=never` on the listbox probably isn't going to do something because that's not the scroll box that's producing that scrollbar.

I'm not quite sure what the actual way to fix it is. Maybe @Vultraz has some ideas.

Reply:
This has been completed - the attribute is `characters_per_line` for the label.